src/components/fx/hover.js

```diff
@@ -1675,38 +1780,27 @@ function alignHoverText(hoverLabels, rotateLabels, scaleX, scaleY) {
         var tx = g.select('text.nums');
         var anchor = d.anchor;
         var horzSign = anchor === 'end' ? -1 : 1;
-        var alignShift = {start: 1, end: -1, middle: 0}[anchor];
-        var txx = alignShift * (HOVERARROWSIZE + HOVERTEXTPAD);
-        var tx2x = txx + alignShift * (d.txwidth + HOVERTEXTPAD);
-        var offsetX = 0;
-        var offsetY = d.offset;
+        var shiftX = getLabelShiftX(d);
+        var offsets = getHoverLabelOffsets(d, rotateLabels);
```

Comment:
Let's store `x` and `y` in variables and continue using `offsetX` and `offsetY` in the code below.
```js
var offset = getHoverLabelOffset(d, rotateLabels);
var offsetX = offset.x;
var offsetY = offset.y;
```

Reply:
Would you mind considering my comment above: https://github.com/plotly/plotly.js/pull/6442/files#r1096117470
That would help reduce the diff and avoid multiple key access.
Thank you!

Reply:
Sure. added changes.